GAE/src/com/gallatinsystems/framework/dao/BaseDAO.java

```diff
@@ -328,10 +341,14 @@ public List filterByUserAuthorizationObjectId(List allObjectsList) {
                 Survey s = (Survey) obj;
                 List<Long> ancestorIds = s.getAncestorIds();
                 if (hasAuthorizedAncestors(ancestorIds, securedObjectIds)) {
-                    authorizedList.add(s);
+                    authorizedSet.add(s);
                 }
             }
         }
+
+        List authorizedList = new ArrayList();
+        authorizedList.addAll(authorizedSet);
```

Comment:
It seems you can pass the set in the constructor to `ArrayList` - https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#ArrayList(java.util.Collection)